### DISCUSSION TITLE
How to use "burpSuite().shutdown()" correctly with burp headless?

### DISCUSSION THREAD

**Discussion by petrabrunner** [PortSwigger/burp-extensions-montoya-api#28] [Q&A] [answered]
Hi, 

I am writing a plugin for burp pro (which is intended to be run in --headless in the context of the plugin). 
and in certain conditions it does not make sense to keep burp up and running - but to terminate the program automatically. 

I am using `MontoyaApi burpApi.burpSuite().shutdown();` for that. 
However I noticed that if used like this the program flow will continue... This makes me assume that the default behavior of shutdown() is to show a prompt to a user. (other possible explanation is of course - that this is a bug)

How can I change that behavior? -> I checked https://github.com/PortSwigger/burp-extensions-montoya-api/blob/main/src/main/java/burp/api/montoya/burpsuite/ShutdownOptions.java  but I was not really able to figure out how to work with PROMPT_USER.

**Comment by Hannah-PortSwigger** [comment, marked as answer]
Hi

If you would like Burp to shut down without prompting the user, you would use `api.burpSuite().shutdown();`.

If you would like it to prompt the user prior to shutting down, you would use `api.burpSuite().shutdown(ShutdownOptions.PROMPT_USER);`.

Is your extension reaching your shutdown statement?

**Reply by petrabrunner** [reply]
Hi, that is interesting... 

yes the extension does make it to that instruction - AND past it.

**Reply by SeanBurnsUK** [reply]
Shutdown is a nonblocking call. It tells burp to shutdown which may take a while (seconds) (Unload all extensions, Close browsers, Close project files, Pause tasks, What for threads to finish, clean up resources etc).

**Reply by petrabrunner** [reply]
ok. so it may be completely normal for some exceptions to be thrown and logged, after the shutdown, depending which state the shutdown of burp is already in. 

-> is there any way to force an immediate shutdown? (the exceptions thrown in my case are actually really confusing, and the logfile is the only way of "seeing" what happened, as burp is tarted in a headless way)

**Reply by Hannah-PortSwigger** [reply]
Could you drop us an email at support@portswigger.net with your code so we can look into this further?

**Reply by petrabrunner** [reply]
will do once I have the time to extract the code a bit
